File: Samples/Graphics/Patterns/C#/Patterns.cs
using System;
using System.Diagnostics;

namespace BitMiracle.Docotic.Pdf.Samples
{
    public static class Patterns
    {
        public static void Main()
        {
            // NOTE:
            // Without a license, the library won't allow you to create or read PDF documents.
            // To get a free time-limited license key, use the form on
            // https://bitmiracle.com/pdf-library/download

            LicenseManager.AddLicenseData("PUT-LICENSE-HERE");

            string pathToFile = "Patterns.pdf";

            using (var pdf = new PdfDocument())
            {
                PdfCanvas canvas = pdf.Pages[0].Canvas;
                canvas.Font = pdf.CreateFont(PdfBuiltInFont.HelveticaBoldOblique);
                canvas.FontSize = 28;

                DrawColoredPattern(pdf);
                DrawUncoloredPattern(pdf);

                pdf.Save(pathToFile);
            }

            Console.WriteLine($"The output is located in {Environment.CurrentDirectory}");

            Process.Start(new ProcessStartInfo(pathToFile) { UseShellExecute = true });
        }

        private static void DrawColoredPattern(PdfDocument pdf)
        {
            PdfTilingPattern pattern = pdf.CreateColoredPattern(5, 5);
            Fill(pattern);

            PdfCanvas canvas = pdf.GetPage(0).Canvas;
            canvas.Brush.Pattern = pattern;
            canvas.DrawString(new PdfPoint(50, 50), "Pattern-filled text");
            canvas.DrawCircle(new PdfPoint(0, 0), 20, PdfDrawMode.FillAndStroke);
        }

        private static void DrawUncoloredPattern(PdfDocument pdf)
        {
            PdfTilingPattern pattern = pdf.CreateUncoloredPattern(5, 5, new PdfRgbColorSpace());
            Fill(pattern);

            PdfCanvas canvas = pdf.GetPage(0).Canvas;
            canvas.Brush.Pattern = pattern;
            canvas.Brush.Color = new PdfRgbColor(0, 255, 0);
            canvas.DrawString(new PdfPoint(50, 150), "Uncolored Pattern colored green");
        }

        private static void Fill(PdfTilingPattern pattern)
        {
            PdfCanvas canvas = pattern.Canvas;
            var red = new PdfRgbColor(255, 0, 0);
            canvas.Brush.Color = red;
            canvas.Pen.Color = red;
            canvas.AppendCircle(new PdfPoint(2, 2), 2);
            canvas.FillAndStrokePath(PdfFillMode.Winding);
        }
    }
}

File: Samples/Graphics/Patterns/VB.NET/Patterns.vb
Imports BitMiracle.Docotic.Pdf

Namespace BitMiracle.Docotic.Pdf.Samples
    Public NotInheritable Class Patterns
        Public Shared Sub Main()
            ' NOTE: 
            ' When used in trial mode, the library imposes some restrictions.
            ' Please visit http://bitmiracle.com/pdf-library/trial-restrictions.aspx
            ' for more information.

            Dim pathToFile As String = "Patterns.pdf"

            Using pdf As New PdfDocument()
                Dim canvas As PdfCanvas = pdf.Pages(0).Canvas
                Dim helvetica As PdfFont = pdf.AddFont(PdfBuiltInFont.HelveticaBoldOblique, False, False)
                canvas.Font = helvetica
                canvas.FontSize = 28

                drawColoredPattern(pdf)
                drawUncoloredPattern(pdf)

                pdf.Save(pathToFile)
            End Using

            Console.WriteLine($"The output is located in {Environment.CurrentDirectory}")

            Process.Start(New ProcessStartInfo(pathToFile) With {.UseShellExecute = True})
        End Sub

        Private Shared Sub drawColoredPattern(ByVal pdf As PdfDocument)
            Dim pattern As PdfTilingPattern = pdf.AddColoredPattern(5, 5)
            fill(pattern)

            Dim canvas As PdfCanvas = pdf.GetPage(0).Canvas
            canvas.Brush.Pattern = pattern
            canvas.DrawString(New PdfPoint(50, 50), "Pattern-filled text")
            canvas.DrawCircle(New PdfPoint(0, 0), 20, PdfDrawMode.FillAndStroke)
        End Sub

        Private Shared Sub drawUncoloredPattern(ByVal pdf As PdfDocument)
            Dim pattern As PdfTilingPattern = pdf.AddUncoloredPattern(5, 5, New PdfRgbColorSpace())
            fill(pattern)

            Dim canvas As PdfCanvas = pdf.GetPage(0).Canvas
            canvas.Brush.Pattern = pattern
            canvas.Brush.Color = New PdfRgbColor(0, 255, 0)
            canvas.DrawString(New PdfPoint(50, 150), "Uncolored Pattern colored green")
        End Sub

        Private Shared Sub fill(ByVal pattern As PdfTilingPattern)
            Dim canvas As PdfCanvas = pattern.Canvas
            Dim red As PdfColor = New PdfRgbColor(255, 0, 0)
            canvas.Brush.Color = red
            canvas.Pen.Color = red
            canvas.AppendCircle(New PdfPoint(2, 2), 2)
            canvas.FillAndStrokePath(PdfFillMode.Winding)
        End Sub
    End Class
End Namespace